src/heroku2Logstash.py

```diff
@@ -0,0 +1,116 @@
+import tornado.ioloop
+import tornado.web
+from tornado import httpclient, gen
+from tornado.httpserver import HTTPServer
+from src.lib import syslogSplitter
+import logging
+import os
+from logging.handlers import SysLogHandler, RotatingFileHandler
+
+stats = {
+    'input': 0,
+    'output': 0,
+    'timeout': 0,
+    'error': 0
+}
+
+
+class HealthCheckHandler(tornado.web.RequestHandler):
+    def get(self):
+        self.set_status(200)
+
+
+class MainHandler(tornado.web.RequestHandler):
+    def set_default_headers(self):
+        self.set_header('Content-Length', '0')
+
+    @gen.coroutine
+    def post(self):
+        payload = self.request.body
+
+        stats['input'] += 1
+        logs = []
+        try:
+            logs = syslogSplitter.split(payload)
+        except Exception as e:
+            logging.getLogger("tornado.application").warning("Error while splitting message ".format(e))
+            logging.getLogger("tornado.application").warning("input headers:{}".format(self.request.headers))
+            logging.getLogger("tornado.application").warning("input payload:{}".format(payload))
+
+        @gen.coroutine
+        def forward(log):
```

Comment:
it is probably intentional to do a closure, but it is far too big to be declared within another function